public/app/plugins/panel/graph3/utils.ts

```diff
@@ -0,0 +1,39 @@
+import { DataFrame, FieldType, getTimeField, sortDataFrame, transformDataFrame } from '@grafana/data';
+
+// very time oriented for now
+export const alignAndSortDataFramesByFieldName = (data: DataFrame[], fieldName: string) => {
+  // normalize time field names
+  // in each frame find first time field and rename it to unified name
```

Comment:
:vomiting_face: mutates the potentially shared object

I think the better solution is to update `seriesToColumns` sol it can join by the first time field regardless of what it is named

Reply:
yes, I like this idea